packages/governance/src/multisig/multisig.cairo

```diff
@@ -0,0 +1,646 @@
+    pub struct CallSalt {
+        #[key]
+        pub id: felt252,
+        pub salt: felt252
+    }
+
+    pub mod Errors {
+        pub const UNAUTHORIZED: felt252 = 'Multisig: unauthorized';
+        pub const NOT_A_SIGNER: felt252 = 'Multisig: not a signer';
+        pub const ALREADY_A_SIGNER: felt252 = 'Multisig: already a signer';
+        pub const ALREADY_CONFIRMED: felt252 = 'Multisig: already confirmed';
+        pub const HAS_NOT_CONFIRMED: felt252 = 'Multisig: has not confirmed';
+        pub const TX_ALREADY_EXISTS: felt252 = 'Multisig: tx already exists';
+        pub const TX_NOT_FOUND: felt252 = 'Multisig: tx not found';
+        pub const TX_NOT_CONFIRMED: felt252 = 'Multisig: tx not confirmed';
+        pub const TX_ALREADY_EXECUTED: felt252 = 'Multisig: tx already executed';
+        pub const ZERO_ADDRESS_SIGNER: felt252 = 'Multisig: zero address signer';
+        pub const ZERO_QUORUM: felt252 = 'Multisig: quorum cannot be 0';
+        pub const QUORUM_TOO_HIGH: felt252 = 'Multisig: quorum > signers';
+    }
+
+    #[embeddable_as(MultisigImpl)]
+    impl Multisig<
+        TContractState, +HasComponent<TContractState>, +Drop<TContractState>
+    > of IMultisig<ComponentState<TContractState>> {
+        /// Returns the current quorum value. The quorum is the minimum number
+        /// of confirmations required to approve a transaction.
+        fn get_quorum(self: @ComponentState<TContractState>) -> u32 {
+            self.Multisig_signers_info.read().quorum
+        }
+
+        /// Returns whether the given `signer` is a registered signer.
+        /// Only registered signers can submit, confirm, or execute transactions.
+        fn is_signer(self: @ComponentState<TContractState>, signer: ContractAddress) -> bool {
+            self.Multisig_is_signer.read(signer)
+        }
+
+        /// Returns the list of all current signers.
+        fn get_signers(self: @ComponentState<TContractState>) -> Span<ContractAddress> {
+            let mut result = array![];
+            let signers_count = self.Multisig_signers_info.read().signers_count;
+            for i in 0..signers_count {
+                result.append(self.Multisig_signers_by_index.read(i));
+            };
+            result.span()
+        }
+
+        /// Returns whether the transaction with the given `id` has been confirmed.
+        /// A confirmed transaction has received the required number of confirmations (quorum).
+        fn is_confirmed(self: @ComponentState<TContractState>, id: TransactionID) -> bool {
+            match self.resolve_tx_state(id) {
+                TransactionState::NotFound => false,
+                TransactionState::Pending => false,
+                TransactionState::Confirmed => true,
+                TransactionState::Executed => true
+            }
+        }
+
+        /// Returns whether the transaction with the given `id` has been confirmed by the specified
+        /// `signer`.
```

Comment:
```suggestion
        /// `signer`.
        ///
        /// NOTE: While this function will still return true if the signer is removed, that confirmation doesn't count when the quorum is checked.
```

cc @andrew-fleming for wording.

Reply:
I'd add a little more context. Something like: "While this function will still return true if the signer is removed after confirming a tx, that confirmation doesn't count when the quorum is checked."

Reply:
And to be super explicit because we can't possible know every which way users will use the component, I'd warn: "This function should therefore not be relied upon for checking legitimate confirmations toward the quorum."

Reply:
Added the suggested detailed version of the note